packages/devtools_app/lib/src/framework/html_framework.dart

```diff
@@ -586,39 +611,113 @@ class HtmlStatusItem {
 }
 
 class HtmlToast extends CoreElement {
-  HtmlToast({this.title, this.message}) : super('div', classes: 'toast') {
+  HtmlToast({this.title, @required this.message})
+      : super('div', classes: 'toast') {
     if (title != null) {
       add(label(text: title));
     }
     add(div(text: message));
-  }
 
-  static const Duration animationDelay = Duration(milliseconds: 500);
-  static const Duration defaultHideDelay = Duration(seconds: 4);
-  static const Duration extendedHideDelay = Duration(seconds: 10);
+    toastAnimator = ToastAnimator(this);
+  }
 
   final String title;
-  @required
+
   final String message;
 
-  void show({Duration hideDelay = defaultHideDelay}) async {
-    await window.animationFrame;
+  ToastAnimator toastAnimator;
 
-    element.style.left = '0px';
+  void show({ToastStyle style = ToastStyle.normal}) async {
+    toastAnimator.show(style: style);
+  }
 
-    Timer(animationDelay, () {
-      Timer(hideDelay, _hide);
-    });
+  @override
+  String toString() => '$title $message';
+}
+
+class HtmlSurveyToast extends CoreElement {
+  HtmlSurveyToast(this._url) : super('div', classes: 'toast') {
+    toastAnimator = ToastAnimator(this);
+
+    layoutVertical();
+    add([
+      div()
+        ..layoutHorizontal()
+        ..add([
+          label(text: 'Help improve DevTools! Take our Q3 ', c: 'toast-title'),
```

Comment:
nit: make more of this message part of the link so it is easier to trigger the survey.